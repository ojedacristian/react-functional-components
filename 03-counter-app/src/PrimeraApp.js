import React from 'react'
import {Fragment} from 'react' // para la 2da app
import PropTypes from 'prop-types'

// Functional Components

const PrimeraApp = ( { saludo, descripcion } )=>{
    return (
        <>
        <h1> { saludo } </h1>
        {/* <pre>{JSON.stringify(saludo,null,3)}</pre> */}
        <p>{ descripcion}</p>
        </>
    );

}


PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

// Tambien se puede agregar al momento de hacer la desestructuracion
// Pero asi si aparece dentro de props en la extension Components de chrome
PrimeraApp.defaultProps = {
    descripcion:'Descripcion por defecto'
}




// Lo mismo pero importando Fragment

export const SegundaApp = ()=>{
    return (
        <Fragment>
        <h1>Hola Mundo</h1>
        <p>Mi Segunda Aplicacion</p>
        </Fragment>
    );

}

export default PrimeraApp;