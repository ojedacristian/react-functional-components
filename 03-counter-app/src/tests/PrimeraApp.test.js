import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import PrimeraApp from '../PrimeraApp'

describe('Probando <PrimeraApp/>', () => {
    
    test('probando', ()=>{
        const saludo = 'Hello'
        const wrapper = shallow(<PrimeraApp saludo={ saludo } descripcion='Descripcion por defecto'/>);
        expect(wrapper).toBe;
    })
})
