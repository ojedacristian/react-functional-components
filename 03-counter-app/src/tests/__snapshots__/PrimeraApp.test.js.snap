// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <PrimeraApp/> probando 1`] = `
<Fragment>
  <h1>
     
    Hello
     
  </h1>
  <p>
    Descripcion por defecto
  </p>
</Fragment>
`;
